rootfs/etc/nginx/lua/lua_ingress.lua

```diff
@@ -2,9 +2,32 @@ local _M = {}
 
 local seeds = {}
 local original_randomseed = math.randomseed
+
+local function get_seed_from_urandom()
+  local seed
+  local frandom, err = io.open("/dev/urandom", "rb")
+  if not frandom then
+    ngx.log(ngx.WARN, 'failed to open /dev/urandom: ', err)
+    return nil
+  end
+
+  local str = frandom:read(4)
+  frandom:close()
+  if not str then
+    ngx.log(ngx.WARN, 'failed to read data from /dev/urandom')
+    return nil
+  end
+
+  seed = 0
+  for i = 1, 4 do
+      seed = 256 * seed + str:byte(i)
+  end
```

Comment:
If I understand this correctly you treat the 4 bytes as a base 256 number (where `str:byte(4)` is the least significant) and then convert it into decimal. Why? Why not simply concatenate the 4 bytes and then use `tonumber` cast it to a decimal? (I'm not suggesting/asking any change, just want to know what's going on - I don't have much experience with random numbers and crypto in general).

Reply:
both solutions are ok, however, if we used the `tonumber`, we need one more function call, and one more table to contact the 4 bytes.
BTW, there is 4 bytes, so we don't need to consider number overflow.

Reply:
This is not a hot code path, I'd be okay to tradeoff performance for readability but reading this again I think treating the bytes we get from `/dev/urandom` as base 256 number when converting to decimal makes more sense since they are bytes.